### DISCUSSION TITLE
Is it possible to log just part of the response body?

### DISCUSSION THREAD

**Discussion by ThiagoFigaro** [apache/apisix#11360] [Q&A]
Im using loki-logger and need to generate statistics using some fields of my response body, but i cant save the whole content because of security/privacy reasons.
Is there any way of using the body-tranformer plugin olny for log purposes, just after the response is integrally delivered to the user?
Or even use regex to erase/replace the fields that i cannot save in loki?
Any ideas, apart of developing my own plugin? :-) tks!

**Comment by juzhiyuan** [comment]
Hi @ThiagoFigaro, I understand you are looking for something like Data Masking. 

If yes, here are two references:

1. https://apisix.apache.org/blog/2023/07/20/data-mask-plugin/
2. https://docs.api7.ai/hub/data-mask

**Reply by ThiagoFigaro** [reply]
Tks for your attention @juzhiyuan 

I need something like data-mask, but apllied to response body. The client must receive the response body without any mask, and i need to log it, but masking or removing some fields.

If i understood correctly, those features you suggested allow to mask only request fields, or mask the response body before it reaches the client.
